pkg/engine/shield/absorb.go

```diff
@@ -15,21 +15,26 @@ func (mgr *Manager) AbsorbDamage(target key.TargetID, damage float64) float64 {
 	// 6. remaining state should be only shields that starting HP > damage
 
 	// placeholder just for some basic event emission
-	for id, shield := range mgr.targets[target] {
+	for _, shield := range mgr.targets[target] {
 		prevHP := shield.HP
 		shield.HP -= damage
 		if shield.HP < 0 {
 			shield.HP = 0
 		}
 		mgr.event.ShieldChange.Emit(event.ShieldChange{
-			ID:     id,
+			ID:     shield.name,
 			Target: target,
 			OldHP:  prevHP,
 			NewHP:  shield.HP,
 		})
 
 		if shield.HP == 0 {
-			mgr.RemoveShield(id, target)
+			mgr.RemoveShield(shield.name, target)
+			damage -= prevHP
+		}
+
+		if shield.HP != 0 {
+			damage = 0
 		}
```

Comment:
This current logic is incorrect shield behavior. For some incoming damage X, that damage X is applied to **all** shields that are currently equipped (say you take 100 damage and have a 200 HP shield and a 150 HP shield. After the hit you will have a 100 HP shield and a 50 HP shield).

Current logic has first shield absorb damage, and the remaining goes to the next and so forth (current logic can also have `damage` go negative).